src/proc/synthetic-stream.cpp

```diff
@@ -92,14 +95,36 @@ namespace librealsense
         {
             return input;
         }
+
+        bool disparity_result_frame = false;
+        bool depth_result_frame = false;
+
+        for (auto f : results)
+        {
+            auto format = f.get_profile().format();
+            if (format == RS2_FORMAT_DISPARITY32 || format == RS2_FORMAT_DISPARITY16)
+                disparity_result_frame = true;
+            if (format == RS2_FORMAT_Z16)
+                depth_result_frame = true;
+        }
+
         std::vector<rs2::frame> original_set;
         if (auto composite = input.as<rs2::frameset>())
-            composite.foreach([&original_set](const rs2::frame& frame) { original_set.push_back(frame); });
+            composite.foreach([&](const rs2::frame& frame)
+            {
+                //switch disparity frames by z16 and vice versa
```

Comment:
This heuristic is an architecture decision that should be documented in depth.  We don't have additional place to store it ,and I suspect we will probably revisit it later on...
e.g. ... - when both depth and disparity frames exist in the frameset, only the target frame will be passed, the other part will be removed... The pros & cons...